src/components/form/HdRange.vue

```diff
@@ -0,0 +1,315 @@
+<template>
+  <div
+    :class="fieldClasses"
+    class="range field"
+  >
+    <input type="range"
+      :name="name"
+      ref="input"
+      :v-model="currentValue"
+      :required="required"
+      :disabled="disabled"
+      :min="minValue"
+      :max="maxValue"
+      :step="rangeStep"
+      @input="updateRangeDecoration"
+      @change="isLegacy ? updateRangeDecoration : null"
+      @focus="handleFocus"
+      @blur="handleBlur"
+    >
+    <div class="range__thumb" ref="thumb">
+      <div class="range__thumb__bullet">
+      </div>
+    </div>
+    <div class="range__decoration">
+      <div class="range__progress" ref="progress">
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+
+export default {
+  name: 'HdRange',
+  props: {
+    name: {
+      type: String,
+      default: 'range',
+    },
+    required: {
+      type: Boolean,
+      default: false,
+    },
+    disabled: {
+      type: Boolean,
+      default: false,
+    },
+    minValue: {
+      type: Number,
+      default: 0,
+    },
+    maxValue: {
+      type: Number,
+      default: 100,
+    },
+    value: {
+      type: Number,
+      default: 0,
+    },
+    rangeStep: {
+      type: Number,
+      default: 1,
+    },
+    isLegacy: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  data() {
+    return {
+      isActive: null,
+    };
+  },
+  computed: {
+    fieldClasses() {
+      return {
+        'field--active': this.isActive,
+        'field--disabled': this.disabled,
+      };
+    },
+  },
+  watch: {
+    currentValue() {
+      this.updateRangeDecoration();
+    },
+    minValue() {
+      this.updateRangeDecoration();
+    },
+    maxValue() {
+      this.updateRangeDecoration();
+    },
+    value(value) {
+      this.$refs.input.value = value;
+      this.updateRangeDecoration();
+    },
+  },
+  mounted() {
+    this.input = this.$refs.input;
+    this.input.value = this.value;
+    this.progress = this.$refs.progress;
+    this.progress.style.width = '1px';
+
+    this.thumb = this.$refs.thumb;
+
+    this.updateRangeDecoration();
+  },
+  methods: {
+    updateRangeDecoration() {
+      const currentValue = this.currentValue();
+      this.$emit('input', parseFloat(currentValue));
+
+      // the percentage of the value, between max and min. I.e. : 15 is the 50% between 10 and 20
+      const valuePercentage = currentValue / ((this.$props.maxValue - this.$props.minValue) / 100);
+      const valuePercentageInputWidthPixels = this.input.offsetWidth * valuePercentage / 100;
+
+      this.progress.style.transform = `scaleX(${valuePercentageInputWidthPixels})`;
+
+      const thumbWidthOffset = this.thumb.offsetWidth * valuePercentage / 100;
+      this.thumb.style.transform = `translateX(${valuePercentageInputWidthPixels - thumbWidthOffset}px)`;
+    },
+    handleFocus() {
+      this.isActive = true;
+    },
+    handleBlur() {
+      this.isActive = false;
+    },
+    currentValue() {
```

Comment:
You're using this method as a modal (`v-modal`)? I'm not sure that will work. A modal needs to be settable and gettable.
I suggest adding `currentValue` as a data property, renaming this method and calling it in the beginning of `updateRangeDecoration()`

Reply:
I was using it to validate the value, but the range element input value does it by itself. So I remvove this.